### PR TITLE
G3 2024 v2 #14525

### PR DESCRIPTION
**Pull request by d22mikwe**
Created functionality to store code examples in the sqlite database tables gitFiles and gitRepo in metadata2.db.
Also created functionality to store the code examples in LenaSYS/courses/$cid/

To test that this works. insert a repository and filepath 
image

When save button is clicked a folder should be created in LenaSYS/courses/ named after the courseid unless the folder already exists. Inside the folder all codeexamples should appear.

Use DB Browser to check metadata2.db and see if all files have been stored in gitFiles and a gitRepo connection has been made.

### CODE REVIEW

**Comment by d22mikwe** [comment]
Talked with william and a different issue is to be made to improve readability, error handling, input handling, commenting and general effectiveness. Also make sure it doesnt overwrite data etc

File: DuggaSys/sectioned.js [hunk 1]
```diff
@@ -4064,6 +4064,49 @@ function fetchGitCodeExamples(courseid){
...
+function storeCodeExamples(cid, codeExamplesContent, githubURL){
+  var base64;
+  var decodedContent = [];
+  var shaKeys = [];
+  var fileNames = [];
+  var fileURL = [];
+  var filePath = [];
+  var downloadURL = [];
+  var fileType = [];
+  for(i = 0; i < codeExamplesContent.length; i++){
```
Comment (a22willi) [inline DuggaSys/sectioned.js:4076]: iterator `i` declared as a global variable.

File: DuggaSys/sectioned.php [hunk 2]
```diff
@@ -681,7 +681,84 @@ function updateGithubDir($pdo, $githubDir, $lid) {
...
+	print_r($codeExamplesContent[0]);
+	print_r($SHA);
+	print_r($fileNames);
+	$WriteFilesSucces = true;
+	$success = true;
+	$path = '../../LenaSYS/courses/'. $cid;
+     // Creates the directory for the corresponding course if it doesnt exist.
+    if (!file_exists($path)) {
+      mkdir($path, 0777, true);
+    }
```
Comment (a22willi) [inline DuggaSys/sectioned.php:740]: Should the directory be created with access for anyone (777)?

File: DuggaSys/sectioned.php [hunk 3]
```diff
@@ -681,7 +681,84 @@ function updateGithubDir($pdo, $githubDir, $lid) {
...
+    //Writes files in folder
+    for($i = 0; $i < count($codeExamplesContent); $i++){
```
Comment (a22willi) [inline DuggaSys/sectioned.php:742]: According to the [php-documentation](https://www.php.net/manual/en/control-structures.for.php) expression 2 of a for loop is evaluated at the beginning of each iteration. Consider moving `count($codeExamplesContent)` outside the for loop.

File: DuggaSys/sectioned.php [hunk 4]
```diff
@@ -681,7 +681,84 @@ function updateGithubDir($pdo, $githubDir, $lid) {
...
+       $WriteFilesSucces = file_put_contents($path . '/' . $fileNames[$i], $codeExamplesContent[$i]);
+	   if($WriteFilesSucces === false){
+         echo "File failed to write";
+		 $success = false;
+	   }else{
+		 echo "File written successfully";
+	   }
+	}
```
Comment (a22willi) [inline DuggaSys/sectioned.php:750]: Fix indentation.

File: DuggaSys/sectioned.php [hunk 5]
```diff
@@ -681,7 +681,84 @@ function updateGithubDir($pdo, $githubDir, $lid) {
...
+	if (!$query->execute()) {
+		$error = $query->errorInfo();
+		echo "Error updating file entries" . $error[2];
+		$errorvar = $error[2];
+		print_r($error);
+		echo $errorvar;
+    }else{
+		return true;
+	}
+}
```
Comment (a22willi) [inline DuggaSys/sectioned.php:700]: Fix indentation.

File: DuggaSys/sectioned.php [hunk 6]
```diff
@@ -681,7 +681,84 @@ function updateGithubDir($pdo, $githubDir, $lid) {
...
+
+//Insert files into gitFiles DB
+function insertIntoSqLiteGitFiles($cid, $fileNames, $filePaths, $fileURLS, $downloadURL, $fileTypes, $SHA){
+	$pdoLite = new PDO('sqlite:../../githubMetadata/metadata2.db');
+	for($i = 0; $i < count($fileNames); $i++){
+		$query = $pdoLite->prepare('INSERT INTO gitFiles (cid, fileName, fileType, fileURL, downloadURL, fileSHA, filePath) VALUES (:cid, :fileName, :fileType, :fileURL, :downloadURL, :fileSHA, :filePath)');
```
Comment (a22willi) [inline DuggaSys/sectioned.php:706]: According to the [php-documentation](https://www.php.net/manual/en/control-structures.for.php) expression 2 of a for loop is evaluated at the beginning of each iteration. Consider moving `count($codeExamplesContent)` outside the for loop if possible.

**Review by a22willi — CHANGES_REQUESTED** [review]

**Comment by a22willi** [comment]
Due to there not being enough time to properly test this PR, I will close it. Create a new branch for next week, and merge this branch with the new week 3 branch.

**Pull request closed without merging**